app/controllers/pet_apps_controller.rb

```diff
@@ -0,0 +1,6 @@
+class PetAppsController < ApplicationController 
+  def create
+    PetApp.create!(application_id: params[:app_id], pet_id: params[:pet_id])
+    redirect_to "/applications/#{params[:app_id]}"
+  end
+end
```

Comment:
Now that you know about flash messages, refrain from using `!` methods in your controllers.